### ISSUE TITLE
Node crashed during consensus

### ISSUE THREAD

**Issue by pocoloko** [idena-network/idena-go#894]
My personal mining node running at home crashed completely while consensus was being reached on today's validation:

```
INFO [01-01|14:12:39.209] After long blocks without ceremonial txs cnt=19
INFO [01-01|14:12:39.209] After long blocks without ceremonial txs shardId=4 cnt=5
INFO [01-01|14:12:39.210] After long blocks without ceremonial txs shardId=3 cnt=5
INFO [01-01|14:12:39.210] After long blocks without ceremonial txs shardId=2 cnt=1
INFO [01-01|14:12:39.210] After long blocks without ceremonial txs shardId=1 cnt=2
INFO [01-01|14:12:39.216] Reached FINAL                            block=0xbbb4b637229303ba8f893fff7b2e3f99095390c66cb13e76d6358ec5b4cab5a1 txs=0
INFO [01-01|14:12:39.217] Node is synchronized                     component=downloader
INFO [01-01|14:12:39.983] no slots for shard, peer will be disconnected peerId=QmWbr6TLE14aMdwBo1L6ndcRsgnL1nii4KhyR2h67BVhJC shardId=3
panic: Value missing for hash 361aeedc459fc8e73497b3c691fddc3c1aacee17f9bf9db7c1109502b2758d53 corresponding to nodeKey 6e361aeedc459fc8e73497b3c691fddc3c1aacee17f9bf9db7c1109502b2758d53

goroutine 5358 [running]:
github.com/cosmos/iavl.(*nodeDB).GetNode(0xc00fd8c300, 0xc00e55bb00, 0x20, 0x20, 0x0)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/nodedb.go:91 +0x6f7
github.com/cosmos/iavl.(*Node).getLeftNode(...)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:423
github.com/cosmos/iavl.(*Node).get(0xc01d99c640, 0xc0119b29f0, 0xc01a3c1788, 0x15, 0x18, 0x15, 0xffffffffffffffff, 0xc0158b2d80, 0xc00fd8c300)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:173 +0x21a
github.com/cosmos/iavl.(*Node).get(0xc003719c20, 0xc0119b29f0, 0xc01a3c1788, 0x15, 0x18, 0x15, 0xffffffffffffffff, 0xc0158b2d80, 0xc00fd8c300)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:173 +0x1af
github.com/cosmos/iavl.(*Node).get(0xc00cf60b40, 0xc0119b29f0, 0xc01a3c1788, 0x15, 0x18, 0x15, 0xffffffffffffffff, 0xc0158b2d80, 0xc00fd8c300)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:173 +0x1af
github.com/cosmos/iavl.(*Node).get(0xc005868fa0, 0xc0119b29f0, 0xc01a3c1788, 0x15, 0x18, 0x15, 0xffffffffffffffff, 0xc0158b2d80, 0xc00fd8c300)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:173 +0x1af
github.com/cosmos/iavl.(*Node).get(0xc010a90fa0, 0xc0119b29f0, 0xc01a3c1788, 0x15, 0x18, 0x15, 0x1, 0xc0158b2d80, 0xc00fd8c300)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:173 +0x1af
github.com/cosmos/iavl.(*Node).get(0xc00cf61cc0, 0xc0119b29f0, 0xc01a3c1788, 0x15, 0x18, 0x15, 0xffffffffffffffff, 0x73, 0xc00fd8c300)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:176 +0x273
github.com/cosmos/iavl.(*Node).get(0xc0050210e0, 0xc0119b29f0, 0xc01a3c1788, 0x15, 0x18, 0x15, 0x1, 0x2f3f468, 0xc00fd8c300)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:173 +0x1af
github.com/cosmos/iavl.(*Node).get(0xc005009360, 0xc0119b29f0, 0xc01a3c1788, 0x15, 0x18, 0x15, 0xffffffffffffffff, 0x9b1e8223301b0e1e, 0x94acecc669461977)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:176 +0x273
github.com/cosmos/iavl.(*Node).get(0xc004e1d180, 0xc0119b29f0, 0xc01a3c1788, 0x15, 0x18, 0xc01a3c1788, 0xd42616bd09b2d11b, 0xc002563c00, 0x0)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:173 +0x1af
github.com/cosmos/iavl.(*ImmutableTree).Get(0xc0119b29f0, 0xc01a3c1788, 0x15, 0x18, 0x15, 0x18, 0x1, 0x18)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/immutable_tree.go:153 +0x5a
github.com/idena-network/idena-go/core/state.(*MutableTree).Get(0xc00a7873e0, 0xc01a3c1788, 0x15, 0x18, 0x0, 0x0, 0x0, 0x0)
        /home/runner/work/idena-go/idena-go/core/state/tree.go:118 +0xb4
github.com/idena-network/idena-go/core/state.(*StateDB).getStateAccount(0xc009a4c8f0, 0x168a31f2bc3f8957, 0x4277014f5fee2f93, 0xf1000000690d4fc6, 0x10)
        /home/runner/work/idena-go/idena-go/core/state/statedb.go:658 +0x207
github.com/idena-network/idena-go/core/state.(*StateDB).GetOrNewAccountObject(0xc009a4c8f0, 0x168a31f2bc3f8957, 0x4277014f5fee2f93, 0x690d4fc6, 0xe3f94edf93a36800)
        /home/runner/work/idena-go/idena-go/core/state/statedb.go:844 +0x3f
github.com/idena-network/idena-go/core/state.(*NonceCache).getAccount(0xc002933f20, 0x168a31f2bc3f8957, 0x4277014f5fee2f93, 0x4f690d4fc6, 0xc0050d1c60)
        /home/runner/work/idena-go/idena-go/core/state/nonce_cache.go:76 +0x37b
github.com/idena-network/idena-go/core/state.(*NonceCache).UnsafeSetNonce(...)
        /home/runner/work/idena-go/idena-go/core/state/nonce_cache.go:67
github.com/idena-network/idena-go/core/state.(*NonceCache).SetNonce(0xc002933f20, 0x168a31f2bc3f8957, 0x4277014f5fee2f93, 0x4f690d4fc6, 0x12)
        /home/runner/work/idena-go/idena-go/core/state/nonce_cache.go:63 +0xa5
github.com/idena-network/idena-go/core/mempool.(*TxPool).put(0xc0010ccff0, 0xc00f1fc3c0, 0xc00e5608a0, 0x3)
        /home/runner/work/idena-go/idena-go/core/mempool/txpool.go:426 +0x256
github.com/idena-network/idena-go/core/mempool.(*TxPool).add(0xc0010ccff0, 0xc00f1fc3c0, 0xc00e5608a0, 0x690d4f00, 0x3, 0x0, 0xc002637d08)
        /home/runner/work/idena-go/idena-go/core/mempool/txpool.go:351 +0x565
github.com/idena-network/idena-go/core/mempool.(*TxPool).AddExternalTxs(0xc0010ccff0, 0x3, 0xc0050d1fb8, 0x1, 0x1, 0x0, 0x0)
        /home/runner/work/idena-go/idena-go/core/mempool/txpool.go:280 +0x2e5
github.com/idena-network/idena-go/core/mempool.(*AsyncTxPool).loop(0xc002ab10c0)
        /home/runner/work/idena-go/idena-go/core/mempool/async_txpool.go:73 +0x157
created by github.com/idena-network/idena-go/core/mempool.NewAsyncTxPool
        /home/runner/work/idena-go/idena-go/core/mempool/async_txpool.go:27 +0x95

```

**Comment by pocoloko** [comment]
Seems this is universal across maybe all nodes? Log from a shared node:

```
panic: Value missing for hash 24466c7bbff9d75f2c625be2e1414c36a14a868b407dbd933cf2d1d190f7212b corresponding to nodeKey 6e24466c7bbff9d75f2c625be2e1414c36a14a868b407dbd933cf2d1d190f7212b

goroutine 126031 [running]:
github.com/cosmos/iavl.(*nodeDB).GetNode(0xc038446cc0, 0xc017622160, 0x20, 0x20, 0x0)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/nodedb.go:91 +0x6f7
github.com/cosmos/iavl.(*Node).getLeftNode(...)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:423
github.com/cosmos/iavl.(*Node).get(0xc00a1410e0, 0xc008671368, 0xc01e4df9c8, 0x15, 0x18, 0x15, 0x1, 0xc019dfbe30, 0xc038446cc0)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:173 +0x21a
github.com/cosmos/iavl.(*Node).get(0xc0049ec3c0, 0xc008671368, 0xc01e4df9c8, 0x15, 0x18, 0x15, 0xffffffffffffffff, 0xc019dfbe30, 0xc038446cc0)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:176 +0x273
github.com/cosmos/iavl.(*Node).get(0xc00d123ae0, 0xc008671368, 0xc01e4df9c8, 0x15, 0x18, 0x15, 0xffffffffffffffff, 0xc019dfbe30, 0xc038446cc0)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:173 +0x1af
github.com/cosmos/iavl.(*Node).get(0xc016ec7220, 0xc008671368, 0xc01e4df9c8, 0x15, 0x18, 0x15, 0x1, 0xc019dfbe30, 0xc038446cc0)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:173 +0x1af
github.com/cosmos/iavl.(*Node).get(0xc012fbf5e0, 0xc008671368, 0xc01e4df9c8, 0x15, 0x18, 0x15, 0xffffffffffffffff, 0x73, 0xc038446cc0)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:176 +0x273
github.com/cosmos/iavl.(*Node).get(0xc012fbec80, 0xc008671368, 0xc01e4df9c8, 0x15, 0x18, 0x15, 0xffffffffffffffff, 0x2f3f468, 0xc038446cc0)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:173 +0x1af
github.com/cosmos/iavl.(*Node).get(0xc012ed81e0, 0xc008671368, 0xc01e4df9c8, 0x15, 0x18, 0x15, 0x1, 0x2ce348c52ae79a42, 0x5e86e0c7b4910ec5)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:173 +0x1af
github.com/cosmos/iavl.(*Node).get(0xc013ec5ae0, 0xc008671368, 0xc01e4df9c8, 0x15, 0x18, 0xc01e4df9c8, 0x25f87ca81b3529d, 0xc00003ec00, 0x0)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:176 +0x273
github.com/cosmos/iavl.(*ImmutableTree).Get(0xc008671368, 0xc01e4df9c8, 0x15, 0x18, 0x15, 0x18, 0x1, 0x18)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/immutable_tree.go:153 +0x5a
github.com/idena-network/idena-go/core/state.(*MutableTree).Get(0xc02e121500, 0xc01e4df9c8, 0x15, 0x18, 0x0, 0x0, 0x0, 0x0)
        /home/runner/work/idena-go/idena-go/core/state/tree.go:118 +0xb4
github.com/idena-network/idena-go/core/state.(*StateDB).getStateAccount(0xc0119873f0, 0xea7df1d419f05fca, 0x89894873f8b37f94, 0x71000000c219b824, 0x10)
        /home/runner/work/idena-go/idena-go/core/state/statedb.go:658 +0x207
github.com/idena-network/idena-go/core/state.(*StateDB).GetOrNewAccountObject(0xc0119873f0, 0xea7df1d419f05fca, 0x89894873f8b37f94, 0xc219b824, 0xefc7a9772e086500)
        /home/runner/work/idena-go/idena-go/core/state/statedb.go:844 +0x3f
github.com/idena-network/idena-go/core/state.(*NonceCache).getAccount(0xc006581a28, 0xea7df1d419f05fca, 0x89894873f8b37f94, 0x4fc219b824, 0xc011863c60)
        /home/runner/work/idena-go/idena-go/core/state/nonce_cache.go:76 +0x37b
github.com/idena-network/idena-go/core/state.(*NonceCache).UnsafeSetNonce(...)
        /home/runner/work/idena-go/idena-go/core/state/nonce_cache.go:67
github.com/idena-network/idena-go/core/state.(*NonceCache).SetNonce(0xc006581a28, 0xea7df1d419f05fca, 0x89894873f8b37f94, 0x4fc219b824, 0x7)
        /home/runner/work/idena-go/idena-go/core/state/nonce_cache.go:63 +0xa5
github.com/idena-network/idena-go/core/mempool.(*TxPool).put(0xc00b1fc870, 0xc0065f46c0, 0xc00fbaf560, 0x3)
        /home/runner/work/idena-go/idena-go/core/mempool/txpool.go:426 +0x256
github.com/idena-network/idena-go/core/mempool.(*TxPool).add(0xc00b1fc870, 0xc0065f46c0, 0xc00fbaf560, 0xc219b800, 0x3, 0x0, 0xc009dba208)
        /home/runner/work/idena-go/idena-go/core/mempool/txpool.go:351 +0x565
github.com/idena-network/idena-go/core/mempool.(*TxPool).AddExternalTxs(0xc00b1fc870, 0x3, 0xc011863fb8, 0x1, 0x1, 0x0, 0x0)
        /home/runner/work/idena-go/idena-go/core/mempool/txpool.go:280 +0x2e5
github.com/idena-network/idena-go/core/mempool.(*AsyncTxPool).loop(0xc00b1d6cb0)
        /home/runner/work/idena-go/idena-go/core/mempool/async_txpool.go:73 +0x157
created by github.com/idena-network/idena-go/core/mempool.NewAsyncTxPool
        /home/runner/work/idena-go/idena-go/core/mempool/async_txpool.go:27 +0x95

```

**Comment by sidenaio** [comment]
Closed as duplicate of #893
